src/history/hs_rec.c

```diff
@@ -612,31 +577,8 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
                  */
                 WT_ASSERT(session, prev_upd->start_ts <= prev_upd->durable_ts);
 
-                /*
-                 * Pop from the out of order timestamp updates stack if the previous update or the
-                 * current update is at the head of the stack. We need to check both cases because
-                 * if there is a tombstone older than the out of order timestamp, we would not pop
-                 * it because we skip the tombstone. Pop it when we are inserting it instead.
-                 *
-                 * Here it is assumed that the out of order update is equal to the oldest update
-                 * among the multiple out of order consecutive updates that have same timestamps.
-                 * For instance, U1@10 -> U2@10 -> U3@10 -> U4@20, U3 which is the oldest update
-                 * will be the out of order update.
-                 */
-                if (out_of_order_ts_upd != NULL &&
-                  (out_of_order_ts_upd == prev_upd || out_of_order_ts_upd == upd)) {
-                    __wt_update_vector_pop(&out_of_order_ts_updates, &out_of_order_ts_upd);
-                    out_of_order_ts_upd = NULL;
-                }
-
-                if (out_of_order_ts_upd != NULL &&
-                  out_of_order_ts_upd->start_ts < prev_upd->start_ts) {
-                    tw.durable_stop_ts = out_of_order_ts_upd->durable_ts;
-                    tw.stop_ts = out_of_order_ts_upd->start_ts;
-                } else {
-                    tw.durable_stop_ts = prev_upd->durable_ts;
-                    tw.stop_ts = prev_upd->start_ts;
-                }
+                tw.durable_stop_ts = prev_upd->durable_ts;
```

Comment:
I don't understand how this case should work U@20->U@0->U@10->U@5? We insert U@5 with (5,10)? I think that is wrong.